src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+function showMessage(message, isError = false) {
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add(isError ? 'error' : 'success');
+  messageDiv.setAttribute('data-qa', 'notification');
+  messageDiv.textContent = message;
+  document.body.appendChild(messageDiv);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const logo = document.querySelector('.logo');
+
+  logo.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => {
+    showMessage(message);
+  })
+  .catch((error) => {
+    showMessage(error.message, true);
+  });
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+secondPromise.then((message) => {
+  showMessage(message);
+});
+
+let rightClick = false;
+let leftClick = false;
+
+const thirdPromise = new Promise((resolve) => {
+  function bothClick() {
+    if (rightClick && leftClick) {
```

Comment:
The logic for resolving the `thirdPromise` when both `rightClick` and `leftClick` are true may not work as expected. Since event listeners are asynchronous, the order of events is not guaranteed. Consider resetting the flags after resolving the promise to avoid multiple resolutions.